bin/plugin/commands/performance.js

```diff
@@ -0,0 +1,228 @@
+ * @property {number} focus Average block selection time.
+ * @property {number} minFocus Min block selection time.
+ * @property {number} maxFocus Max block selection time.
+ */
+/**
+ * @typedef WPFormattedPerformanceResults
+ *
+ * @property {string} load Load Time.
+ * @property {string} domcontentloaded DOM Contentloaded time.
+ * @property {string} type Average type time.
+ * @property {string} minType Minium type time.
+ * @property {string} maxType Maximum type time.
+ * @property {string} focus Average block selection time.
+ * @property {string} minFocus Min block selection time.
+ * @property {string} maxFocus Max block selection time.
+ */
+
+/**
+ * Computes the average number from an array numbers.
+ *
+ * @param {number[]} array
+ * @return {number} Average.
+ */
+function average( array ) {
+	return array.reduce( ( a, b ) => a + b ) / array.length;
+}
+
+/**
+ * Computes the median number from an array numbers.
+ *
+ * @param {number[]} array
+ * @return {number} Median.
+ */
+const median = ( array ) => {
+	const mid = Math.floor( array.length / 2 ),
+		numbers = [ ...array ].sort( ( a, b ) => a - b );
+	return array.length % 2 !== 0
+		? numbers[ mid ]
+		: ( numbers[ mid - 1 ] + numbers[ mid ] ) / 2;
+};
+
+/**
+ * Rounds and format a time passed in milliseconds.
+ *
+ * @param {number} number
+ * @return {string} Formatted time.
+ */
+function formatTime( number ) {
+	const factor = Math.pow( 10, 2 );
+	return Math.round( number * factor ) / factor + ' ms';
+}
+
+/**
+ * Curate the raw performance results.
+ *
+ * @param {WPRawPerformanceResults} results
+ * @return {WPPerformanceResults} Curated Performance results.
+ */
+function curateResults( results ) {
+	return {
+		load: average( results.load ),
+		domcontentloaded: average( results.domcontentloaded ),
+		type: average( results.type ),
+		minType: Math.min( ...results.type ),
+		maxType: Math.max( ...results.type ),
+		focus: average( results.focus ),
+		minFocus: Math.min( ...results.focus ),
+		maxFocus: Math.max( ...results.focus ),
+	};
+}
+
+/**
+ * Runs the performance tests on a given branch.
+ *
+ * @param {string} performanceTestDirectory Path to the performance tests' clone.
+ * @param {string} environmentDirectory Path to the plugin environment's clone.
+ * @param {string} branch Branch name.
+ *
+ * @return {Promise<WPFormattedPerformanceResults>} Performance results for the branch.
+ */
+async function getPerformanceResultsForBranch(
+	performanceTestDirectory,
+	environmentDirectory,
+	branch
+) {
+	log( '>> Fetching the ' + formats.success( branch ) + ' branch' );
+	await git.checkoutRemoteBranch( environmentDirectory, branch );
+
+	log( '>> Building the ' + formats.success( branch ) + ' branch' );
+	await runShellScript(
+		'npm install && npm run build',
+		environmentDirectory
+	);
+
+	log(
+		'>> Running the test on the ' + formats.success( branch ) + ' branch'
+	);
+	const results = [];
+	for ( let i = 0; i < 3; i++ ) {
```

Comment:
> Or is it just an arbitrary number to try toward some confidence of trusting the output? 

This is the reason, I think cache could create wrong numbers that's why I trust multiple runs better.

For the order, I agree that it's better to switch but the problem is that it will make the command a lot slower unless we have separate wp-env environments entirely. I might consider it separately.